plugin-gradle/README.md

```diff
@@ -203,6 +203,33 @@ org.gradle.jvmargs=--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAME
 ```
 This is a workaround to a [pending issue](https://github.com/diffplug/spotless/issues/834).
 
+### palantir-java-format
+
+[homepage](https://github.com/palantir/palantir-java-format). [changelog](https://github.com/palantir/palantir-java-format/releases).
+```gradle
+spotless {
+  java {
+    palantirJavaFormat()
+    // optional: you can specify a specific version and/or switch to GOOGLE/AOSP style
+    //   and/or use custom group artifact (you probably don't need this)
+    palantirJavaFormat('2.9.0').style("PALANTIR").groupArtifact('com.google.googlejavaformat:google-java-format')
```

Comment:
I don't think it makes sense to have `groupArtifact` support. It was added in #944 for someone exactly like you - they customized GJF to better support lambdas, except they maintained the same package names so that they didn't need their own step.  It maybe also doesn't make sense to have `style` support? Is anyone going to use `palantir` step but not the `palantir` style?

The GJF step has picked up a fair amount of cruft over the years because people have wanted to add their specific little thing to it. I am happy to merge a PR which either

1. adds a little more cruft to GJF to support Palantir, e.g. `googleJavaFormat('2.10.0').style('PALANTIR').groupArtifact('com.palantir.javaformat:palantir-java-format').package('com.palantir.javaformat')` which could be refactored as `palantirJavaFormat()` which could call `GoogleJavaFormatStep.defaultVersionPalantir()` for `JvmSupport`, etc.
2. OR creates a new step which removes all the cruft to the bare minimum API that you are actually using right now

My problem with this PR is that the palantir step is starting its life on day 1 with all the accumulated complexity of the GJF step without a clear usecase for why any of that complexity is needed anymore. If you really want to speciate away from GJF, we're happy to support that by giving you your own step to develop your own features and API. But if you don't plan to develop your own API independent of GJF, then you should probably stay compatible with GJF. There's lot of new language features coming to Java and GJF is definitely not abandoned...

Happy to merge 1 or 2, and this just needs a small bit of work to become that.

Reply:
New commits clearly take step 2, looks good.